src/server/system_services/account_server.js

```diff
@@ -803,15 +803,34 @@ async function update_external_connection(req) {
         throw new RpcError('INVALID_CREDENTIALS', `Credentials are not valid ${name}`);
     }
 
+    let set_obj;
+    if (azure_log_access_keys?.azure_tenant_id?.unwrap()) {
```

Comment:
why checking for azure_tenant_id existence and not for azure_log_access_keys?

Reply:
try adding +optional to the operator code

Reply:
I think that here we should validate the values of `azure_log_access_keys`. we already have the [schema](https://github.com/noobaa/noobaa-core/blob/845073892c4768ec6e0edf593b0baadd20686a85/src/api/common_api.js#L754C1-L754C1) that enforces all the required fields. if these fields are invalid we should throw an error.
in the operator we should make sure we call this API correctly